test/cases/orchestration.py

```diff
@@ -325,6 +329,30 @@ def header_args():
     return fields, body
 
 
+def add_services_in_nginx_etc_hosts(services):
+    """When we added build/test support on ARM64 by using the ARM64 execution
+    environment in CircleCI, we started seeing intermittent delays in DNS
+    lookups, always about five seconds.
+
+    This function uses `getent` to look up the IP address of each `docker
+    compose` service, and then associates the address with the service name by
+    adding a line to /etc/hosts in the nginx service container. This way, the
+    default `gethostbyname()` resolver used by nginx will not use DNS."""
+    script = f"""
+    for service in {" ".join(f"'{service}'" for service in services)}; do
+        getent hosts "$service" >>/etc/hosts
```

Comment:
First time I encounter `getent`, interesting. I thought you would use `docker inspect`.